File: app/src/main/java/com/swc/parsedweatherviewerkotlin/utils/CommonUtils.java
package com.swc.parsedweatherviewerkotlin.utils;

import com.swc.parsedweatherviewerkotlin.model.WeatherElement;
import com.swc.parsedweatherviewerkotlin.model.WeatherRow;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class CommonUtils {

    private static final String TAG = "CommonUtils";

    /**
     *
     * @param url
     * @param query
     * @return
     */
    public static List<WeatherRow> getTodayWeatherTable(String url, String query) {

        List<WeatherRow> weatherTable = new ArrayList<>();
        try {
            Document doc = Jsoup.connect(url).get();
            Element table = doc.select(query).first();

            for (Element row : table.select("tr")) {
                //for each row. get region, morning, afternoon table elements
                WeatherElement amWeather = null, pmWeather = null;
                Element region = row.select("th").first();

                Elements icons = row.getElementsByClass("icon");
                Elements summaries = row.getElementsByClass("nm");
                Elements rains = row.getElementsByClass("rain");
                Elements temps = row.getElementsByClass("temp");

                if (icons.size() > 1 && summaries.size() > 1 && rains.size() > 1 && temps.size() > 1){
                    amWeather = new WeatherElement(summaries.get(0).text(), rains.get(0).text(), temps.get(0).text(), icons.get(0).select("img").first().attr("abs:src"));
                    pmWeather = new WeatherElement(summaries.get(1).text(), rains.get(1).text(), temps.get(1).text(), icons.get(1).select("img").first().attr("abs:src"));
                    weatherTable.add(new WeatherRow(region.text(), amWeather, pmWeather));
                }
            }
        } catch (IOException e) {

        }

        return weatherTable;

    }
}

File: app/src/main/java/com/swc/parsedweatherviewerkotlin/utils/CommonUtils.kt
package com.swc.parsedweatherviewerkotlin.utils

import com.swc.parsedweatherviewerkotlin.model.WeatherElement
import com.swc.parsedweatherviewerkotlin.model.WeatherRow
import org.jsoup.Jsoup
import java.io.IOException
import java.util.ArrayList

/**
 * Created by sangwonc on 2018. 6. 11..
 */

object CommonUtils {

    val TAG = "CommonUtils"

    /**
     *
     * @param url
     * @param query
     * @return
     */
    fun getTodayWeatherTable(url: String, query: String): List<WeatherRow> {

        val weatherTable = ArrayList<WeatherRow>()
        try {
            val doc = Jsoup.connect(url).get()
            val table = doc.select(query).first()

            for (row in table.select("tr")) {
                //for each row. get region, morning, afternoon table elements
                var amWeather: WeatherElement? = null
                var pmWeather: WeatherElement? = null
                val region = row.select("th").first()

                val icons = row.getElementsByClass("icon")
                val summaries = row.getElementsByClass("nm")
                val rains = row.getElementsByClass("rain")
                val temps = row.getElementsByClass("temp")

                if (icons.size > 1 && summaries.size > 1 && rains.size > 1 && temps.size > 1) {
                    amWeather = WeatherElement(summaries[0].text(), rains[0].text(), temps[0].text(), icons[0].select("img").first().attr("abs:src"))
                    pmWeather = WeatherElement(summaries[1].text(), rains[1].text(), temps[1].text(), icons[1].select("img").first().attr("abs:src"))
                    weatherTable.add(WeatherRow(region.text(), amWeather, pmWeather))
                }
            }
        } catch (e: IOException) {

        }

        return weatherTable

    }
}
